cogeo_mosaic/backends/base.py

```diff
@@ -64,3 +69,82 @@ def write(self):
     @abc.abstractmethod
     def update(self):
         """Update existing MosaicJSON on backend."""
+
+    def _update_quadkey(self, quadkey: str, dataset: List[str]):
+        """Update quadkey list."""
+        self.mosaic_def.tiles[quadkey] = dataset
+
+    def _update_metadata(self, bounds: List[float], version: str):
```

Comment:
Since we are updating the model directly it would be nice to re-validate on assignment using the `validate_assignment` [model config](https://pydantic-docs.helpmanual.io/usage/model_config/)

```python
class MosaicJSON(BaseModel):
    ...

    class Config:
        validate_asignment = True
```